.github/workflows/release-workflow.yml

```diff
@@ -0,0 +1,58 @@
+name: Changesets Release
+
+on:
+  workflow_dispatch:
+
+jobs:
+  CreateRelease:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    permissions:
+      id-token: write
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0  # Fetch all history for all branches and tags
+        token: ${{ secrets.IMJS_ADMIN_GH_TOKEN }}
+
+    - name: Install pnpm
+      uses: pnpm/action-setup@v2
+      with:
+        version: 8
+        run_install: false
+
+    - name: Use Node.js 18
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18
+        cache: "pnpm"
+
+    - name: Install dependencies
+      run: pnpm install
+
+    - name: ESLint
+      run: pnpm run lint
+
+    - name: Typecheck
+      run: pnpm run typecheck
+
+    - name: Check changed-elements-react unit test coverage
+      run: npm run test:cover --prefix ./packages/changed-elements-react
+
+    - name: Run Audit
+      run: pnpm audit
+
+    - name: Create release PR or publish to npm
+      uses: changesets/action@v1
+      with:
+        publish: pnpm run build:components && changeset publish
+        title: Release packages [publish docs]
+        commit: Release packages [publish docs]
+        baseBranch: master
```

Comment:
is this a valid config?
wouldnt this be driven by whats in the changesets config?